test/e2e/helpers.go

```diff
@@ -284,15 +285,31 @@ func logCheckpoint(specTimes map[string]time.Time) {
 	}
 }
 
+// getMachinesInCluster returns a list of all machines in the given cluster.
+// This is copied from CAPI's test/framework/cluster_proxy.go.
+func getMachinesInCluster(ctx context.Context, c framework.Lister, namespace, name string) (*clusterv1.MachineList, error) {
```

Comment:
what about machine pools? Do we want to add that here too @devigned ?

Reply:
Probably. There should now be similar funcs for MachinePools.

Reply:
I added support here for MachinePools and created a helper function to collect all the SSH info for the nodes in a cluster, whether they are control plane or agent, VM or VMSS.

One thing I'm curious about is that we anticipated needing port 50001 for VMSS instances, but in practice it seems to be good old port 22 everywhere...

Reply:
maybe port 50001 was an AKS Engine thing? Although I would expect the VMSS node pools to go through port 22 for the first control plane ssh but only control plane VMSS nodes (not supported currently) to require a different port.